nisystemlink/clients/artifact/_artifact_client.py

```diff
@@ -0,0 +1,64 @@
+"""Implementation of ArtifactClient"""
+
+from typing import Optional
+
+from httpcore import Response
+from nisystemlink.clients import core
+from nisystemlink.clients.core._uplink._base_client import BaseClient
+from nisystemlink.clients.core._uplink._methods import get, post
+from nisystemlink.clients.core.helpers._iterator_file_like import IteratorFileLike
+from uplink import Part, Path, response_handler
+
+from . import models
+
+
+class ArtifactClient(BaseClient):
+    def __init__(self, configuration: Optional[core.HttpConfiguration] = None):
+        """Initialize an instance.
+
+        Args:
+            configuration: Defines the web server to connect to and information about
+                how to connect. If not provided, the
+                :class:`HttpConfigurationManager <nisystemlink.clients.core.HttpConfigurationManager>`
+                is used to obtain the configuration.
+
+        Raises:
+            ApiException: if unable to communicate with the  Notebook execution Service.
+        """
+        if configuration is None:
+            configuration = core.HttpConfigurationManager.get_configuration()
+
+        super().__init__(configuration, base_path="/ninbartifact/v1/")
+
+    @post("artifacts")
+    def upload_artifact(

```

Comment:
It would be better to create a wrapper to help with Type hinting as `Part` does not indicate the type.